internal/engine/actions/remediate/pull_request/lib.go

```diff
@@ -0,0 +1,352 @@
+
+type modificationRegistry map[string]modificationConstructor
+
+func newModificationRegistry() modificationRegistry {
+	return make(map[string]modificationConstructor)
+}
+
+func (mr modificationRegistry) register(name string, constructor modificationConstructor) {
+	mr[name] = constructor
+}
+
+func (mr modificationRegistry) registerBuiltIn() {
+	mr.register(minderContentModification, newContentModification)
+	mr.register(minderFrizbeeTagResolve, newFrizbeeTagResolveModification)
+}
+
+func (mr modificationRegistry) getModification(
+	name string,
+	params *modificationConstructorParams,
+) (fsModifier, error) {
+	constructor, ok := mr[name]
+	if !ok {
+		return nil, fmt.Errorf("unknown modification type: %s", name)
+	}
+
+	return constructor(params)
+}
+
+var _ fsModifier = (*contentModification)(nil)
+
+type contentModification struct {
+	fsChangeSet
+	prCfg *pb.RuleType_Definition_Remediate_PullRequestRemediation
+}
+
+var _ modificationConstructor = newContentModification
+
+func newContentModification(
+	params *modificationConstructorParams,
+) (fsModifier, error) {
+	// validate
+	if params.prCfg == nil {
+		return nil, fmt.Errorf("pull request config cannot be nil")
+	}
+
+	if len(params.prCfg.Contents) == 0 {
+		return nil, fmt.Errorf("pull request config contents cannot be empty")
+	}
+
+	for i := range params.prCfg.Contents {
+		cnt := params.prCfg.Contents[i]
+		if cnt.Path == "" {
+			return nil, fmt.Errorf("pull request config contents path cannot be empty")
+		}
+		if cnt.Content == "" {
+			return nil, fmt.Errorf("pull request config contents content cannot be empty")
+		}
+	}
+
+	return &contentModification{
+		prCfg: params.prCfg,
+		fsChangeSet: fsChangeSet{
+			fs: params.bfs,
+		},
+	}, nil
+}
+
+func (ca *contentModification) createFsModEntries(
+	_ context.Context,
+	params interfaces.ActionsParams,
+) error {
+	entries, err := ca.prConfigToEntries()
+	if err != nil {
+		return fmt.Errorf("cannot create PR entries: %w", err)
+	}
+
+	data := map[string]interface{}{
+		"Params":  params.GetRule().Params.AsMap(),
+		"Profile": params.GetRule().Def.AsMap(),
+	}
+	for i := range entries {
+		entry := entries[i]
+		content := new(bytes.Buffer)
+
+		if err := entry.contentTemplate.Execute(content, data); err != nil {
+			return fmt.Errorf("cannot execute content template (index %d): %w", i, err)
+		}
+		entry.Content = content.String()
+	}
+
+	ca.entries = entries
+	return nil
+
+}
+
+func (ca *contentModification) prConfigToEntries() ([]*fsEntry, error) {
+	entries := make([]*fsEntry, len(ca.prCfg.Contents))
+	for i := range ca.prCfg.Contents {
+		cnt := ca.prCfg.Contents[i]
+
+		contentTemplate, err := util.ParseNewTextTemplate(&cnt.Content, fmt.Sprintf("Content[%d]", i))
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse content template (index %d): %w", i, err)
+		}
+
+		mode := ghModeNonExecFile
+		if cnt.Mode != nil {
+			mode = *cnt.Mode
+		}
```

Comment:
Want to explicitly strip off the leading numbers if the length is >4 and the last 4 are an octal mode?  e.g.

```suggestion
		}
		// We used to use Git modes, which have a bunch of high-order bits.  Strip them out.
		if len(mode) > 4 && mode[len(mode)-4] == '0' {
			mode = mode[len(mode)-4:]
		}
```

Reply:
since we reverted to using git modes again, this comment is obsolete